src/litserve/api.py

```diff
@@ -63,6 +63,13 @@ def predict(self, x, **kwargs):
         pass
 
     def _unbatch_no_stream(self, output):
+        if isinstance(output, str):
+            warnings.warn(
+                "The 'predict' method returned a string instead of a list of predictions. "
+                "When batching is enabled, 'predict' should return a list of predictions. "
+                "To avoid unexpected behavior, ensure 'predict' returns a list of predictions, "
+                "or implement 'LitAPI.unbatch' correctly."
```

Comment:
Predict should alway return outputs in correct format.
```suggestion
                "To avoid unexpected behavior, ensure 'predict' returns a list of predictions."
```